hal/src/main/native/sim/Notifier.cpp

```diff
@@ -83,18 +84,58 @@ void WakeupNotifiers() {
   });
 }
 
+void WaitNotifiers() {
+  std::unique_lock ulock(notifiersWaiterMutex);
+  wpi::SmallVector<HAL_NotifierHandle, 8> waiters;
+
+  // Wait for all Notifiers to hit HAL_WaitForNotifierAlarm()
+  notifierHandles->ForEach([&](HAL_NotifierHandle handle, Notifier* notifier) {
+    std::scoped_lock lock(notifier->mutex);
+    if (notifier->active && !notifier->waitingForAlarm) {
+      waiters.emplace_back(handle);
+    }
+  });
+  if (waiters.size() == 0) {
```

Comment:
This isn't necessary to special case... it falls through the loop below.  If you want to special case it, use empty() rather than size == 0.